packages/modals/src/elements/DrawerModal/DrawerModal.tsx

```diff
@@ -0,0 +1,192 @@
+import mergeRefs from 'react-merge-refs';
+import { CSSTransition } from 'react-transition-group';
+import { ThemeContext } from 'styled-components';
+import { useModal } from '@zendeskgarden/container-modal';
+import { useDocument } from '@zendeskgarden/react-theming';
+import { useFocusVisible } from '@zendeskgarden/container-focusvisible';
+import { ModalsContext } from '../../utils/useModalContext';
+import {
+  StyledDrawerModalBackdrop,
+  StyledDrawerModal,
+  StyledDrawerModalFooter,
+  StyledDrawerModalFooterItem
+} from '../../styled';
+
+import { Header } from './Header';
+import { Body } from './Body';
+import { Close } from './Close';
+
+interface IStaticDrawerModalExport<T, P>
+  extends ForwardRefExoticComponent<PropsWithoutRef<P> & RefAttributes<T>> {
+  Header: typeof Header;
+  Body: typeof Body;
+  Close: typeof Close;
+  Footer: typeof StyledDrawerModalFooter;
+  FooterItem: typeof StyledDrawerModalFooterItem;
+}
+
+export interface IDrawerModalProps extends HTMLAttributes<HTMLDivElement> {
+  /**
+   * Determines whether the `DrawerModal` is opened or closed
+   */
+  isOpen?: boolean;
+  /**
+   * HTML attributes to spread onto backdrop element
+   */
+  backdropProps?: any;
+  /**
+   * Callback when a close action has been completed.
+   * Can be triggered from the backdrop and Close icon.
+   * @param {Object} event - DOM event that triggered the close action
+   */
+  onClose?: (event: KeyboardEvent | MouseEvent) => void;
+  /**
+   * The root ID to use for descendants. A unique ID is created if none is provided.
+   **/
+  id?: string;
+  /**
+   * The DOM element where the drawer modal will be rendered to
+   **/
+  appendToNode?: Element;
+  /**
+   * Determines whether to apply keyboard focus on the drawer modal on mount
+   */
+  focusOnMount?: boolean;
+  /**
+   * Determines whether to return keyboard focus to the element that triggered the drawer modal
+   */
+  restoreFocus?: boolean;
+}
+
+export const DrawerModal = forwardRef<
+  HTMLDivElement,
+  IDrawerModalProps & HTMLAttributes<HTMLDivElement>
+>(
+  (
+    { id, isOpen, onClose, backdropProps, appendToNode, focusOnMount, restoreFocus, ...props },
+    ref
+  ) => {
+    const modalRef = useRef<HTMLDivElement | null>(null);
+    const theme = useContext(ThemeContext);
+    const environment = useDocument(theme);
+
+    useFocusVisible({ scope: modalRef, relativeDocument: modalRef.current });
+
+    const {
+      getTitleProps,
+      getCloseProps,
+      getContentProps,
+      getBackdropProps,
+      getModalProps
+    } = useModal({
+      id,
+      modalRef,
+      focusOnMount,
+      restoreFocus,
+      environment,
+      onClose
+    });
+
+    useEffect(() => {
+      if (!environment) {
+        return undefined;
+      }
+
+      const bodyElement = environment.querySelector('body');
+
+      if (bodyElement && isOpen) {
+        const previousBodyOverflow = bodyElement.style.overflow;
+
+        bodyElement.style.overflow = 'hidden';
```

Comment:
I'm still tripping on this. But I'm willing to suspend belief in the interest of getting the component out into people's hands. For me: modal + transparent background ≠ locked scrolling. Let's see if there is kick-back.